modin/experimental/core/execution/native/implementations/hdk_on_native/calcite_builder.py

```diff
@@ -106,6 +107,51 @@ def gen_reduce_expr(self):
             """
             pass
 
+    class CompoundAggregate2(CompoundAggregate):
```

Comment:
can't we add support for multiple arguments to the existing `CompundAggregate` class?

Reply:
I've moved some code into this class to be reused in #6528. Both TopK and Quantile aggregates have a similar logic which requires creation of the argument column.

Reply:
Okay, can this class be named somewhat differently then? 'CompoundAggregate2' looks strange (like a movie sequel :D).

Maybe, `MultipleExpressionsAgg`?